cedar-wasm/src/lib.rs

```diff
@@ -1,8 +1,252 @@
+        Ok(context) => context,
+        Err(err) => {
+            return json!({
+                "code": 104,
+                "message": format!("[ContextErr]: {}", err),
+            })
+            .to_string();
+        }
+    };
+
+    let policies = match PolicySet::from_str(policies_str) {
+        Ok(policies) => policies,
+        Err(err) => {
+            return json!({
+                "code": 105,
+                "message": format!("[PoliciesErr]: {}", err),
+            })
+            .to_string();
+        }
+    };
+
+    let entities = match Entities::from_json_str(entities_str, None) {
+        Ok(entities) => entities,
+        Err(err) => {
+            return json!({
+                "code": 106,
+                "message": format!("[EntitiesErr]: {}", err),
+            })
+            .to_string();
+        }
+    };
+
+    let request = match Request::new(Some(principal), Some(action), Some(resource), context, None) {
+        Ok(request) => request,
+        Err(err) => {
+            return json!({
+                "code": 107,
+                "message": format!("[RequestErr]: {}", err),
+            })
+            .to_string();
+        }
+    };
+
+    let authorizer = Authorizer::new();
+    let response = authorizer.is_authorized(&request, &policies, &entities);
+
+    // change response to string
+    let decision = response.decision();
+    let diagnostics = response.diagnostics();
+
+    let _reasons = diagnostics.reason();
+    let mut reasons = HashSet::new();
+    for reason in _reasons {
+        reasons.insert(reason.to_string());
+    }
+
+    let _errors = diagnostics.errors();
+    let mut errors = Vec::new();
+    for err in _errors {
+        let error = err.to_string();
+        errors.push(error);
+    }
+
+    json!({
+        "code": 0,
+        "data": {
+            "decision": decision,
+            "reasons": reasons,
+            "errors": errors,
+        }
+    })
+    .to_string()
+}
+
+#[wasm_bindgen(js_name = "validate")]
+pub fn validate(schema_str: &str, policy_str: &str) -> String {
+    let schema = match Schema::from_str(schema_str) {
+        Ok(schema) => schema,
+        Err(err) => {
+            return json!({
+                "code": 201,
+                "message": format!("[SchemaErr]: {}", err),
+            })
+            .to_string();
+        }
+    };
+
+    let policy = match PolicySet::from_str(policy_str) {
+        Ok(policy) => policy,
+        Err(err) => {
+            return json!({
+                "code": 202,
+                "message": format!("[PolicyErr]: {}", err),
+            })
+            .to_string();
+        }
+    };
+
+    let validator = Validator::new(schema);
+
+    let result = validator.validate(&policy, ValidationMode::default());
+
+    json!({
+        "code": 0,
+        "data": format!("{}", result),
+    })
+    .to_string()
+}
+
+#[wasm_bindgen(js_name = "policyToJson")]
+pub fn policy_to_json(policy_str: &str) -> String {
```

Comment:
The JSON policy stuff is being addressed in this PR https://github.com/cedar-policy/cedar/pull/616/files

You can take a look at that to see how we're exporting rich input and output types to typescript using TSify